File: Java/src/_0171_Excel_Sheet_Column_Number/Solution.java
package _0171_Excel_Sheet_Column_Number;

/**
 * Date: 2021/6/29 14:19
 */
public class Solution {

    public int titleToNumber(String s) {
        char[] cs = s.toCharArray();
        int res = 0;
        for (char c : cs) res = res * 26 + (c - 'A' + 1);
        return res;
    }

    public static void main(String[] args) {

        System.out.println((new Solution()).titleToNumber("A"));  // 1
        System.out.println((new Solution()).titleToNumber("AB"));  // 28
        System.out.println((new Solution()).titleToNumber("ZY"));  // 701
    }
}

File: Python/_0171_Excel_Sheet_Column_Number/Solution.py
# coding=utf-8
# Date: 2021/6/29 14:27


# 执行用时：32 ms, 在所有 Python3 提交中击败了97.89%的用户
# 内存消耗：14.9 MB, 在所有 Python3 提交中击败了56.53%的用户
class Solution:
    def titleToNumber(self, s: str) -> int:
        res = 0
        for c in s:
            res = res * 26 + (ord(c) - ord('A') + 1)
        return res


if __name__ == "__main__":
    print(Solution().titleToNumber("A"))  # 1
    print(Solution().titleToNumber("AB"))  # 28
    print(Solution().titleToNumber("ZY"))  # 701
